### PR TITLE
Fix 12 Boring tests involving NULL-SHA ciphersuites

### PR DESCRIPTION
**Pull request by mattcaswell**
The Boring runner attempts to enable the NULL-SHA ciphersuite using the
cipherstring "DEFAULT:NULL-SHA". However in OpenSSL DEFAULT permanently
switches off NULL ciphersuites, so we fix this up to be "ALL:NULL-SHA"
instead. We can't change the runner so we have to change the shim to
detect this.

##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/ossl_shim/ossl_config.json [hunk 1]
```diff
@@ -36,22 +36,10 @@
...
-        "TLS11-NULL-SHA-server":"Test failure - reason unknown",
-        "TLS1-NULL-SHA-client":"Test failure - reason unknown",
-        "TLS11-NULL-SHA-client":"Test failure - reason unknown",
-        "TLS11-NULL-SHA-LargeRecord":"Test failure - reason unknown",
-        "TLS12-NULL-SHA-client":"Test failure - reason unknown",
-        "TLS12-NULL-SHA-server":"Test failure - reason unknown",
-        "TLS12-NULL-SHA-LargeRecord":"Test failure - reason unknown",
-        "TLS11-NULL-SHA-BadRecord":"Test failure - reason unknown",
-        "TLS12-NULL-SHA-BadRecord":"Test failure - reason unknown",
+        "DTLS1-NULL-SHA-server":"Test failure - reason unknown",
```
Comment (ekasper) [inline test/ossl_shim/ossl_config.json]: Interesting, the DTLS tests still fail?

**Comment by levitte** [comment]
@mattcaswell, it would be good if you have `[extended tests]` in your commit messages for these kinds of tests, so we can see them in action in Travis.  Granted, that will prolongue the build time, but...

File: test/ossl_shim/ossl_config.json (hunk 1 above)
Comment (mattcaswell) [inline test/ossl_shim/ossl_config.json]: Yes, some other problem I suspect

**Comment by mattcaswell** [comment]
@levitte, ok I'll remember that for the next one ;-)

**Comment by levitte** [comment]
I would have liked to see it for this one, but I guess I piped up too late...  ah well...

**Pull request closed without merging**